### ISSUE TITLE
LibJS: false assertion failure in Float64ArrayConstructor::construct

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/serenity#5076]
Found with Fuzzilli.
Source:
```js
const v2 = new Float64Array(RegExp);
```
Trace:
```
USERSPACE(28) ASSERTION FAILED: false
../Userland/Libraries/LibJS/Runtime/TypedArray.cpp:160
CrashDaemon(20): --- Backtrace for thread #0 (TID 28) ---
CrashDaemon(20): 0x020e83e3: [libc.so] __assertion_failed +0x98 (Syscall.h:449)
CrashDaemon(20): 0x0229ea98: [libjs.so] JS::Float64ArrayConstructor::construct(JS::Function&) +0xc6 (TypedArray.cpp:160)
CrashDaemon(20): 0x022a28ee: [libjs.so] JS::VM::construct(JS::Function&, JS::Function&, AK::Optional<JS::MarkedValueList>, JS::GlobalObject&) +0x314 (VM.cpp:252)
CrashDaemon(20): 0x02226ac6: [libjs.so] JS::CallExpression::execute(JS::Interpreter&, JS::GlobalObject&) const +0x7ee (Optional.h:114)
CrashDaemon(20): 0x02226028: [libjs.so] JS::VariableDeclaration::execute(JS::Interpreter&, JS::GlobalObject&) const +0xb2 (AST.cpp:1592)
CrashDaemon(20): 0x0223362f: [libjs.so] JS::Interpreter::execute_statement(JS::GlobalObject&, JS::Statement const&, JS::ScopeType) +0x10f (Interpreter.cpp:173)
CrashDaemon(20): 0x0221f246: [libjs.so] JS::Program::execute(JS::Interpreter&, JS::GlobalObject&) const +0x56 (AST.cpp:108)
CrashDaemon(20): 0x02232e24: [libjs.so] JS::Interpreter::run(JS::GlobalObject&, JS::Program const&) +0x2e4 (VM.h:127)
CrashDaemon(20): 0x0233a1ff: [/bin/js] parse_and_run(JS::Interpreter&, AK::StringView const&) +0x20f (js.cpp:467)
CrashDaemon(20): 0x02336a05: [/bin/js] .L1330 +0x14c (js.cpp:581)
CrashDaemon(20): 0x02336caf: [/bin/js] _start +0x42 (crt0_shared.cpp:60)
```

**Comment by linusg** [comment]
https://github.com/SerenityOS/serenity/blob/13d7c09125f8eec703d0a43a9a87fc8aa08f7319/Userland/Libraries/LibJS/Runtime/TypedArray.cpp#L141-L142

Not a bug, just not implemented yet. :)

(desired outcome is a `Float64Array` of `RegExp`'s `length`, i.e. 2, containing `+RegExp[0]` and `+RegExp[1]`, i.e. `NaN`)

**Comment by linusg** [comment]
This was implemented (and therefore fixed) in #6383.
